website/addons/s3/templates/s3_node_settings.mako

```diff
@@ -77,3 +81,17 @@
         <p data-bind="html: message, attr.class: messageClass"></p>
     </div>
 </div>
+
+ <script>
+    // Make s3 settings variables globally accessible for relevant JS
+    <%
+      from website.addons.s3.settings import BUCKET_LOCATIONS, ENCRYPT_UPLOADS_DEFAULT
+      from website.util.sanitize import safe_json as sjson
+    %>
+    window.contextVars = $.extend(true, {}, window.contextVars, {
+        s3Settings: {
+            encryptUploads: ${ ENCRYPT_UPLOADS_DEFAULT | sjson, n },
+            bucketLocations: ${ BUCKET_LOCATIONS | sjson, n }
```

Comment:
It might make sense to include `encryptUploads` (and any other configuration that needs to be accessed from both Python and JS in the JSON file you created. That way, we have just one place where that configuration lives, and we completely get rid of `window.contextVars` here.

settings.json

```json
{
    "bucketLocations": {...},
    "defaultBucketLocations": ...
    "encryptUploads": ...
}
```